packages/dx-demo-shell/src/demo-viewer/source-code.jsx

```diff
@@ -34,42 +37,61 @@ export class SourceCode extends React.PureComponent {
       height: 'auto',
       gutters: ['CodeMirror-linenumbers', 'CodeMirror-foldgutter'],
     });
-    this.foldBlockStartLines.forEach((lineNumber) => {
-      editor.foldCode(CodeMirror.Pos(lineNumber, 0));
-    });
-    this.importantLines.forEach((lineNumber) => {
-      editor.addLineClass(lineNumber, 'background', 'CodeMirror-important-line');
-    });
+    this.codeMirror.setValue(source);
+    this.applySpecialCodeOptions(foldBlockStartLines, importantLines);
+  }
+
+  componentDidUpdate() {
+    const { source, foldBlockStartLines, importantLines } = this.prepareSourceCode();
```

Comment:
Extract these three lines to a separated method. Call it in `componentDidMount` then.